File: backend/src/main/java/com/ghanamilitaryquiz/model/Achievement.java
package com.ghanamilitaryquiz.model;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.time.LocalDateTime;

@Entity
@Table(name = "achievements")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Achievement {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(length = 255)
    private String icon;

    @Column(name = "criteria_type", nullable = false, length = 50)
    private String criteriaType;

    @Column(name = "criteria_value", nullable = false)
    private Integer criteriaValue;

    @Column(nullable = false)
    private Integer points = 0;

    @Column(name = "badge_color", length = 20)
    private String badgeColor;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
}

File: frontend/src/types/achievement.ts
// Achievement types
export interface Achievement {
  id: number;
  name: string;
  description?: string;
  icon?: string;
  criteriaType: string;
  criteriaValue: number;
  points: number;
  badgeColor?: string;
  isActive: boolean;
  createdAt: string;
}

export interface UserAchievement {
  id: number;
  userId: number;
  achievementId: number;
  earnedAt: string;
  achievement?: Achievement;
}

export interface CreateAchievementRequest {
  name: string;
  description?: string;
  icon?: string;
  criteriaType: string;
  criteriaValue: number;
  points: number;
  badgeColor?: string;
}

export interface UpdateAchievementRequest extends Partial<CreateAchievementRequest> {
  isActive?: boolean;
}
